### DISCUSSION TITLE
Terraform update to 1.4.4 is unsuccessful

### DISCUSSION THREAD

**Discussion by carmelo0x63** [Homebrew/discussions#4388] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 931327df1f30a3a5b1aa07657c0d1070e61988f9
Last commit: 7 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 112f1aed2d485570bd3c9fd37cf8571cd39ceaf1
Core tap last commit: 4 weeks ago
Core tap branch: master
Core tap JSON: 07 Apr 08:08 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.qMg2NhQJ1t/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi all,
I seem to be unable to upgrade Terraform from 1.4.3-dev (?) to 1.4.4 on macOS 13.3. `terraform version` yields:
```
Terraform v1.4.3-dev
on darwin_arm64

Your version of Terraform is out of date! The latest version
is 1.4.4. You can update by downloading from https://www.terraform.io/downloads.html
```

But `brew upgrade terraform` reports the following instead: `Warning: terraform 1.4.4 already installed`
I've tried removing the package and installing it again to no avail.

Additionally:
```
$ which terraform
/opt/homebrew/bin/terraform
```
and
```
$ ls -l /opt/homebrew/bin/terraform
lrwxr-xr-x  1 *** ***  39 Apr  6 20:10 /opt/homebrew/bin/terraform -> ../Cellar/terraform/1.4.4/bin/terraform
```
`/opt/homebrew/Cellar/terraform/1.4.4/` is the only directory on my disk. No other versions seem to exist.
What could be the reason for this discrepancy and how to upgrade Terraform to the latest and greatest?

Thanks,
Carmelo

**Comment by SMillerDev** [comment, marked as answer]
https://github.com/Homebrew/homebrew-core/issues/127453
https://github.com/Homebrew/homebrew-core/issues/127407

**Comment by carmelo0x63** [comment]
Summary of the threads above: "this is not a Homebrew/core issue. Someone has already reported this to Terraform at https://github.com/hashicorp/terraform/issues/32963, and according to https://github.com/hashicorp/terraform/issues/32963#issuecomment-1492577372, this will be fixed in the next release"

Many thanks @SMillerDev for your response!
